### PR TITLE
documentation: remove out-of-date

### PR DESCRIPTION
**Pull request by stasmihailov**
- #5553: Documentation on \<intercept-url filters="..."\> is outdated
- #6602: Spring Security 5 docs references distribution zip, which does not exist

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@stasmihailov Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security&pullRequestId=6603)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-security&pullRequestId=6603) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by jzheaux** [comment]
@stasmihailov thanks for the cleanup!

Can you help me understand why you want to remove the Contacts and Tutorial sections? These samples are still in the repo under: https://github.com/spring-projects/spring-security/tree/master/samples/xml

**Comment by stasmihailov** [comment]
> @stasmihailov thanks for the cleanup!
> 
> Can you help me understand why you want to remove the Contacts and Tutorial sections? These samples are still in the repo under: https://github.com/spring-projects/spring-security/tree/master/samples/xml

Okay, it was mentioned in #6602 that files are missing and I myself haven't ever found them in sources. I will remove that commit

**Comment by jzheaux** [comment]
Yes, that makes sense to change it to refer to the GitHub links to the samples instead of the zip file.

**Comment by pivotal-issuemaster** [comment]
@stasmihailov Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security&pullRequestId=6603)!

**Comment by stasmihailov** [comment]
@pivotal-issuemaster no problem

File: docs/manual/src/docs/asciidoc/_includes/preface/samples.adoc [hunk 1]
```diff
@@ -2,9 +2,7 @@
 
 [[sample-apps]]
 == Sample Applications
-There are several sample web applications that are available with the project.
-To avoid an overly large download, only the "tutorial" and "contacts" samples are included in the distribution zip file.
-The others can be built directly from the source which you can obtain as described in <<get-source,the introduction>>.
+There are several sample web applications which are shipped with project's <<community-source,source code>> (they can be found in https://github.com/spring-projects/spring-security/tree/master/samples/xml[samples/xml/] subdirectory of root project).
```
Comment (jzheaux) [inline docs/manual/src/docs/asciidoc/_includes/preface/samples.adoc]: Would you please update this in the following way:

```suggestion
There are several {gh-samples-url}[sample web applications] that ship with the project's <<community-source,source code>>. You can find the ones described below in the ({gh-samples-url}/xml[samples/xml/] subdirectory of the root project).
```

**Comment by jzheaux** [comment]
@stasmihailov Sorry this has taken a bit for me to get back to you.

You'll notice that one of the tests isn't passing in your PR, and the reason for that is that the code tests to make sure the documentation matches the contents of the xsd.

Would you please do the following as well and include them in your `filters` commit?

1. Remove the corresponding `filters` directive in `spring-security-5.2.rnc`
2. Locally run `./gradlew :spring-security-config:rncToXsd`
3. Add `spring-security-5.2.rnc` and `spring-security-5.2.xsd` in your commit that removes `filters` from the documentation. 

As a double-check, I'd recommend running `./gradlew clean build integrationTest` to make sure that everything passes (just in case I'm missing something).

This will make the xsd and the documentation match, so then the tests will pass.

So, in the end, there will be two commits: One to remove `filters`, and another to update the Samples documentation.

**Comment by stasmihailov** [comment]
@jzheaux deal. Will do this, thanks for assistance!

**Comment by jzheaux** [comment]
Great, @stasmihailov, thanks again! In preparation for merging, would you please squash your `filters`-related commits so that there are now only two: One for `filters` work and the other for the updated samples documentation?

It looks like you might need to do a quick rebase with `master`, too.

Once that is cleaned up, we should be good to merge.

**Comment by jzheaux** [comment]
@stasmihailov The commits still aren't quite right. Right now, the PR has two commits, but the filters work is spread across both of them. And the samples work is combined with the filters work.

Would you please have the commits aligned so that the filter work is in one commit and the samples work is in another commit? The reason for this is in [the contribution guidelines](https://github.com/spring-projects/spring-security/blob/master/CONTRIBUTING.md#squash-commits):

> Use git rebase --interactive, git add --patch and other tools to "squash" multiple commits into _atomic changes._

In this case, for example, updating the filters documentation is one atomic task, so there would be [one commit for that like](https://github.com/spring-projects/spring-security/blob/master/CONTRIBUTING.md#format-commit-messages):

```
Update filters documentation

Fixes: gh-5553
````

And in that commit would be all the changes relating to updating the doc, the rnc, and the xsd. There wouldn't be anything in that commit about updating the samples documentation.

Then, there would be another commit like:

```
Update samples documentation
```

Since there is no associated ticket for that, you could leave the `Fixes` off of that commit message.

Also, when you did the rebase, I think you didn't take the changes from `master`, as the samples commit [attempts to change some `https` links back to `http`](https://github.com/spring-projects/spring-security/pull/6603/files#diff-b7c41f4896d68549fc725ec17cf129eeR84). Could you please take a look at that?

**Comment by stasmihailov** [comment]
@jzheaux okay, ill take a deeper look into guidelines and then will fix commits

**Comment by jzheaux** [comment]
Thanks, @stasmihailov! This is now merged into `master`.

**Pull request merged**